WordPress/src/main/java/org/wordpress/android/ui/reader/repository/ReaderPostRepository.kt

```diff
@@ -1,453 +1,372 @@
+                }
+                UNCHANGED -> {
+                    Log.i(javaClass.simpleName, "***=> is unchanged")
+                }
+                FAILED -> {
+                    Log.i(javaClass.simpleName, "***=> is failed")
+                    // todo: what are we doing about failures?
+                }
+            }
+        }
+    }
+
+    private fun isCurrentTag(tag: ReaderTag): Boolean {
+        return ReaderTag.isSameTag(tag, readerTag)
+    }
+
+    @Subscribe(threadMode = MAIN)
+    fun onEventMainThread(event: SearchPostsStarted) {
+        Log.i(javaClass.simpleName, "***=> TODO")
+    }
+
+    @Subscribe(threadMode = MAIN)
+    fun onEventMainThread(event: SearchPostsEnded) {
+        Log.i(javaClass.simpleName, "***=> TODO")
+    }
+
+    companion object {
+        private const val EXCLUDE_TEXT_COLUMN = true
+        private const val MAX_ROWS = ReaderConstants.READER_MAX_POSTS_TO_DISPLAY
+    }
+
+    class Factory
+    @Inject constructor(
+        private val contextProvider: ContextProvider,
+        private val eventBusWrapper: EventBusWrapper,
+        @Named(BG_THREAD) private val bgDispatcher: CoroutineDispatcher,
+        @Named(IO_THREAD) private val ioDispatcher: CoroutineDispatcher
+    ) {
+        fun create(readerTag: ReaderTag): ReaderPostRepository {
+            return ReaderPostRepository(
+                    contextProvider,
+                    eventBusWrapper,
+                    bgDispatcher,
+                    ioDispatcher,
+                    readerTag
+            )
+        }
+    }
+}
+
+// todo: annmarie - these may or may not stay
+sealed class ReaderRepositoryAction(val action: UpdateAction) {
+    class Discover(val interests: List<String>, action: UpdateAction) :
+            ReaderRepositoryAction(action)
+
+    class PostsForTag(action: UpdateAction) : ReaderRepositoryAction(action)
+
+    class PostsForBlog(action: UpdateAction) : ReaderRepositoryAction(action)
+
+    class PostsForFeed(action: UpdateAction) : ReaderRepositoryAction(action)
+
+    override fun toString(): String {
+        return "${this.javaClass.simpleName}($action)"
+    }
+}
+
+// todo: annmarie - decide if we need to include the action that was requested
+sealed class ReaderRepositoryCommunication {
+    object Success : ReaderRepositoryCommunication()
+    sealed class Error : ReaderRepositoryCommunication() {
+        object NetworkUnavailable : Error()
+        class ReaderRepositoryException(val exception: Exception) : Error()
+    }
+
+    override fun toString(): String {
+        return "${this.javaClass.simpleName})"
+    }
+}
+
+// todo: Annmarie - where should this class live and the listener
+/** ReactiveMutableLiveData is a handy extension of MutableLiveData.
+ * The main purpose is to monitor onActive and onInactive
+ *  situations because they are unreachable using straight up LiveData
+ */
+class ReactiveMutableLiveData<T>(private val onReactiveListener: OnReactiveListener) :
```

Comment:
I want to break out these classes `ReactiveMutableLiveData` and the listeners, but I don't know what package they should be in. Should I drop them in Reader -> Repository?